### PR TITLE
Adding Invalid IP Address Check

### PR DESCRIPTION
**Pull request by singhabh27**
## Checklist

- [ ] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [ ] The title of the PR is clear and informative
- [ ] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [ ] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
        - {Please put the link here}
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

File: src/Migrate/custom/Set-AzMigrateServerReplication.ps1 [hunk 1]
```diff
@@ -635,6 +635,10 @@ function Set-AzMigrateServerReplication {
                             $updateNic.TargetStaticIPAddress = $null
                         }
                         else {
+                            $isValidIpAddress = [ipaddress]::TryParse($matchingUserInputNic.TargetStaticIPAddress,[ref][ipaddress]::Loopback)
```
Comment (prsadhu-ms-idc) [inline src/Migrate/custom/Set-AzMigrateServerReplication.ps1:638]: PS C:\Users\prsadhu> [ipaddress]::TryParse("4",[ref][ipaddress]::Loopback)
True

Is this a valid scenario?

Comment (singhabh27) [reply src/Migrate/custom/Set-AzMigrateServerReplication.ps1:638]: Yes, it will treat it as 0.0.0.4 . Same behavior what we kept in service

Comment (prsadhu-ms-idc) [reply src/Migrate/custom/Set-AzMigrateServerReplication.ps1:638]: Should this be here, or in the other cmdlet where we set NIC details.

Comment (prsadhu-ms-idc) [reply src/Migrate/custom/Set-AzMigrateServerReplication.ps1:638]: I feel it should be in the other cmdlet.

**Pull request merged**